core/templates/dev/head/components/score_ring/ScoreRingDirective.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the animated score ring.
+ */
+
+oppia.directive('scoreRing', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {
+        getScore: '&score',
+      },
+      bindToController: {},
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/components/score_ring/score_ring_directive.html'),
+      controllerAs: '$ctrl',
+      controller: ['$scope', '$window',
+        function($scope, $window) {
+          var setScore = function(percent) {
+            const offset = circumference - percent / 100 * circumference;
+            circle.style.strokeDashoffset = offset;
+          };
+
+          const circle = document.querySelector('.score-ring__circle');
+          const radius = circle.r.baseVal.value;
+          const circumference = radius * 2 * Math.PI;
+          circle.style.strokeDasharray = `${circumference} ${circumference}`;
+          circle.style.strokeDashoffset = circumference;
+
+          angular.element($window).bind('load', function() {
```

Comment:
Hm why do you need this? Can't we just call setScore($scope.getScore()) directly, or using $timeout, without the window.bind('load') stuff? (In general, setting things on $window is a pattern we try to avoid because it's basically "global" scope, and I'm not sure if, in this case, it could end up conflicting with other things that are attached to 'load' as well.)

Reply:
So the reason I used window.load is because the animation part of it doesn't end up showing if the page takes time to load. By the time the page loads we just see a filled ring (up till the score), we don't see it filling. I tried a bunch of time outs, sometimes they work and sometimes they don't - it really depends on how long it takes for the page to load. That's why I thought of doing it on window.load itself. I think $timeout has the same problem - I don't know what the right amount of time delay I should use for the animation to show in every case without there being a visible delay in the animation for fast loads.

Reply:
Ah ok, makes sense, thanks for the explanation. Could you please add a comment in the code to explain what's going on here, for the benefit of future devs? That would provide good context fo rthe future -- I think there are finer-grained controls in Angular 2 for hooking into the lifecycle of a directive, so maybe we can use those once we migrate.

Reply:
Done